code/game/machinery/computer/law.dm

```diff
@@ -28,19 +42,70 @@
 
 
 /obj/machinery/computer/aiupload/attackby(obj/item/O as obj, mob/user as mob, params)
-	if(istype(O, /obj/item/aiModule))
-		if(!current)//no AI selected
-			to_chat(user, "<span class='danger'>No AI selected. Please chose a target before proceeding with upload.")
-			return
-		var/turf/T = get_turf(current)
-		if(!atoms_share_level(T, src))
-			to_chat(user, "<span class='danger'>Unable to establish a connection</span>: You're too far away from the target silicon!")
-			return
+	if(!istype(O, /obj/item/aiModule))
+		return
+	if(!current)//no AI selected
+		to_chat(user, "<span class='danger'>No AI selected. Please choose a target before proceeding with upload.</span>")
+		return
+	var/turf/T = get_turf(current)
+	if(!atoms_share_level(T, src))
+		to_chat(user, "<span class='danger'>Unable to establish a connection</span>: You're too far away from the target silicon!</span>")
+		return
+	if(!emagged) //non-emag law change
 		var/obj/item/aiModule/M = O
 		M.install(src)
 		return
+	else
+		if(world.time < cooldown)
+			to_chat(user, "<span class='danger'>The program seems to have frozen. It will need some time to process.</span>")
+			return
+		do_sparks(5, TRUE, src)
+		countlaws()
+		if(!emag_ion_check())
+			emag_inherent_law()
+		return
 	return ..()
 
+/obj/machinery/computer/aiupload/proc/countlaws()
+	foundlaws = 0
+	for(var/datum/ai_law/law in current.laws.inherent_laws)
+		foundlaws++
+
+/obj/machinery/computer/aiupload/proc/emag_ion_check()
+	var/emag_law = new /datum/ai_law/inherent(generate_ion_law()).law
+	if(!length(current.laws.ion_laws))
+		if(prob(20))  // 20% chance to generate an ion law if none exists
+			current.add_ion_law(generate_ion_law())
+			cooldown = world.time + EMAG_COOLDOWN
+			return TRUE
+	else //10% chance to overwrite a current ion
+		if(prob(10))
+			current.laws.ion_laws[1].law = emag_law
+			cooldown = world.time + EMAG_COOLDOWN
+			log_and_message_admins("has given [current] the ion law: [current.laws.ion_laws[1].law].")
+			return TRUE
+		else
+			return FALSE
+
+/obj/machinery/computer/aiupload/proc/emag_inherent_law()
+	if(!foundlaws)
+		return
+	else
```

Comment:
No need for "else" here, if it has foundlaws then the proc will run normally.

Reply:
done